core/vm/interpreter/impl/interpreter_impl.cpp

```diff
@@ -162,22 +170,51 @@ namespace fc::vm::interpreter {
     return false;
   }
 
-  outcome::result<Result> CachedInterpreter::interpret(
-      const IpldPtr &ipld, const Tipset &tipset) const {
-    // TODO: TipsetKey from art-gor
-    common::Buffer key;
-    for (auto &cid : tipset.cids) {
-      OUTCOME_TRY(encoded, cid.toBytes());
-      key.put(encoded);
+  namespace {
+    const common::Buffer &getBadTipsetMarker() {
+      static const uint8_t z[] = {0x00};
+      static const common::Buffer kBadTipsetMark(
+          gsl::span<const uint8_t>(z, 1));
```

Comment:
```suggestion
      static const common::Buffer kBadTipsetMark{0x00};
```
Maybe remove `getBadTipsetMarker`, just leave const

Reply:
Better, cbor encode/decode `boost::optional<Result>`, null encodes to one byte